api/model/src/main/java/org/projectnessie/model/UDF.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.annotation.JsonView;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import java.util.Map;
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import org.immutables.value.Value;
+import org.projectnessie.model.ser.Views;
+
+@Value.Immutable
+@JsonSerialize(as = ImmutableUDF.class)
+@JsonDeserialize(as = ImmutableUDF.class)
+@JsonTypeName("UDF")
+public abstract class UDF extends IcebergContent {
+
+  @NotBlank
+  @jakarta.validation.constraints.NotBlank
+  @NotNull
+  @jakarta.validation.constraints.NotNull
+  public abstract String getSqlText();
+
+  @Nullable
+  @jakarta.annotation.Nullable // TODO this is currently undefined in Iceberg
+  public abstract String getDialect();
+
+  @Override
+  public Type getType() {
+    return Type.UDF;
+  }
+
+  @Deprecated
+  @Nullable
+  @jakarta.annotation.Nullable
+  @JsonInclude(Include.NON_NULL)
+  @JsonView(Views.V1.class)
+  // Left here in case an old Nessie client sends this piece of information.
+  // To be removed when API v1 gets removed.
+  public abstract Map<String, Object> getMetadata();
```

Comment:
Not sure I understand how this relates to API versions :thinking: If the client uses new jars, won't it be able to sent normal `UDF` objects over API v1 _and_ v2 (whatever it chooses)?

If the client uses old jars, it will not see this method anyway.